Sources/WalletConnectRelay/Dispatching.swift

```diff
@@ -84,28 +102,51 @@ final class Dispatcher: NSObject, Dispatching {
     }
 
     func connect() throws {
-        try socketConnectionHandler.handleConnect()
+        try socketConnectionHandler?.handleConnect()
     }
 
     func disconnect(closeCode: URLSessionWebSocketTask.CloseCode) throws {
-        try socketConnectionHandler.handleDisconnect(closeCode: closeCode)
+        try socketConnectionHandler?.handleDisconnect(closeCode: closeCode)
+    }
+    
+    private func setUpSocketConnection() {
+        setUpSocket()
+        setUpWebSocketSession()
+        setUpSocketConnectionObserving()
+    }
+    
+    private func setUpSocket() {
+        let socket = socketFactory.create(with: relayUrlFactory.create(
+            host: relayHost,
+            projectId: projectId
+        ))
+        socket.request.addValue(EnvironmentInfo.userAgent, forHTTPHeaderField: "User-Agent")
+        self.socket = socket
```

Comment:
why we need to recreate `socket` instance? Is it possible to just change `request` property?